pkg/master/client_ca_hook.go

```diff
@@ -137,7 +138,25 @@ func writeConfigMap(client coreclient.ConfigMapsGetter, name string, data map[st
 		return err
 	}
 
-	existing.Data = data
-	_, err = client.ConfigMaps(metav1.NamespaceSystem).Update(existing)
+	// Reset JSON marshalling for empty slices
```

Comment:
why not `reflect.DeepEqual(existing.Data, data)`?

Reply:
It seems that when the object is being unmarshalled, any of the keys that contain slices (`requestHeaderSliceKeys`), become unmarshalled as `null` rather than `[]`, yet the `data` object has them as `[]` due to the JSON serialization above.

This simply sets those back to what they would be if they had come straight from the JSON serialization and then does the `DeepEqual`

Reply:
configmap data contain literal strings, not slices... it does not interpret those string values. as long as we are consistently invoked, the serialization should not change.

Reply:
Indeed, it seems I've misunderstood the behaviour here, through not really understanding what the tests were doing when I first wrote this.

I've pushed another commit that removes all this extra stuff that isn't needed